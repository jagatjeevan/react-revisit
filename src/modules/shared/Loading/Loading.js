import React from "react";
import "./loading.scss";

const Loading = () => (
  <div className="loading" data-testid="loading">
    Loading Component ...
  </div>
);

export default Loading;
