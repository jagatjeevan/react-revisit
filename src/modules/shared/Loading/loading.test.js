import React from "react";
import { cleanup, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Loading from "./loading";

describe("Loading Component", () => {
  afterEach(cleanup);

  test("it should mount", () => {
    render(<Loading />);
    const loading = screen.getByTestId("loading");
    expect(loading).toBeInTheDocument();
  });

  test("takes a snapshot", () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment(<Loading />)).toMatchSnapshot();
  });
});
