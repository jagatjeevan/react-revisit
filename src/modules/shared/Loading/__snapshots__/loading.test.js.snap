// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component takes a snapshot 1`] = `
<DocumentFragment>
  <div
    class="loading"
    data-testid="loading"
  >
    Loading Component ...
  </div>
</DocumentFragment>
`;
